server/src/main/java/org/opensearch/search/lookup/LeafSearchLookup.java

```diff
@@ -87,4 +90,16 @@ public void setDocument(int docId) {
         sourceLookup.setSegmentAndDocument(ctx, docId);
         fieldsLookup.setDocument(docId);
     }
+
+    public Object getTermFrequency(
```

Comment:
I think this function does not belong here,  may be have a dedicated `LeadTermFrequencyLookup` class?